server/plugins/webapp/index.js

```diff
@@ -103,14 +116,20 @@ function makeRouteHandler(options, userContent) {
         ? `<link rel="manifest" href="${manifest}" />`
         : "";
       const css = bundleCss();
-      const cssLink = css ? `<link rel="stylesheet" href="${css}" />` : "";
+      const cssLink = css && !criticalCSS
+        ? `<link rel="stylesheet" href="${css}" />`
+        : "";
       return `${manifestLink}${cssLink}`;
     };
 
     const makeBodyBundles = () => {
       const js = bundleJs();
+      const css = bundleCss();
+      const cssLink = css && criticalCSS
+      ? `<link rel="stylesheet" href="${css}" />`
+      : "";
```

Comment:
We want to load the stylesheet after the body content if there is critical CSS, so it doesn't block rendering